src/Security/Guard/JwtGuardAuthenticator.php

```diff
@@ -36,27 +41,28 @@ public function __construct(Auth0Service $auth0Service)
 
     /**
      * {@inheritdoc}
+     *
+     * @param Request $request Symfony representation of the HTTP request message.
+     *
+     * @return boolean
      */
+    // phpcs:ignore
     public function supports(Request $request)
     {
-        return $request->headers->has('Authorization') &&
-               strpos($request->headers->get('Authorization'), 'Bearer') === 0;
```

Comment:
Hello @evansims, this seems to cause an issue on my side.
We have a secondary firewall using http basic auth for securing the swagger.
The missing condition here considering the authenticator valid for any request cause an error if there is no valid Bearer token in the Authorization header.

Reply:
Yes, this code should definitely still be part of the authenticator as it will now conflict with any other loaded authenticator in a Symfony application.

Reply:
Hey Quentin, this was an intentional change to ensure that both JwtGuardAuthenticator and JWTAuthenticator behaved in identical/predictable ways; the expectation was from our present documentation that we shouldn't be outright denying access but identifying the user as IS_AUTHENTICATED_ANONYMOUSLY, so the client application could handle how to deal with that situation as appropriate for its needs. As it was, outright denying the connection means there is no opportunity for the client application to intervene with say a login prompt. What are your thoughts?

Reply:
The JWTAuthenticator is deprecated for a while now as is also mentioned in the class docblock. If the guard authenticator can't find anything in the request to authenticate with it shouldn't try to authenticate anyway. I believe this is also mentioned within the Symfony documentation.

Reply:
From my understanding, if the `supports` method return `true`, then you must be able to both get a user and check it's credentials, or the access would be denied. Even if you return an anonymous user at the end of the `getUser` method the next one `checkCredential` will fail without a proper JWT from Auth0.
Moreover, it seems that the firewall allowing anonymous user would already do the wanted behavior, allowing anonymous to access the firewall pages.

Reply:
@niels-nijens It is deprecated by Symfony, but we support more than the most recent versions of Symfony, and it therefore remains and we continue to include it for legacy applications that rely on that functionality.

@renrhaf `checkCredential` intentionally does not fail if a JWT is not present; it will hand off an anonymous user to the application. In my tests, failing within `supports` will actually block the firewall from permitting anonymous users even when configured to do so, but I'll take another look at this and see if I was mistaken

Reply:
@evansims Of course, I understand that multiple versions are supported. What I meant to say is that both systems are different and the behavior could therefore also be different.

Besides my earlier argument; If I remember correctly, by not implementing this code you're not able to differentiate between unauthenticated and forbidden. We have functional tests for this behavior in our applications, so I will try to run them with the updated package tomorrow. 🙂

Reply:
@niels-nijens And I respect and appreciate the feedback here, of course! I'm just not of the mind that offering two approaches to the same problem (JWTAuthenticator and JwtGuardAuthenticator) but having those approaches behave in dramatically different ways is a positive approach, from a developer experience standpoint. I want to limit opportunities to introduce confusion as much as humanly possible. However, if there are technical circumstances here that should be addressed, I'm all for fixing those! I just want to ensure we're not unnecessarily stepping on toes and robbing host applications of freedom to do reasonable decision-making for themselves where it makes sense.

I would definitely anticipate this change would require some tweaking to tests to accommodate the change in approach, but I'd be interested to hear your results. Thanks.

Reply:
@evansims the problem in my case is that I have the following value in the `Authorization` header : `Basic YWRtaW46I3pXViU1cG50TGpy` because I also use http basic auth for my Swagger UI.
![image](https://user-images.githubusercontent.com/690629/111691637-e5b4a500-8804-11eb-96f6-dc75d2035eb2.png)
The call to `decodeJWT` will fail and throw an exception, thus I can not access some pages I should be able to access anonymously.

Reply:
Gotcha @Renrhaf, thanks for the additional details there! That does look like a situation we'll need to handle better. I'll dig into this tomorrow and see what can be done to address that